plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php

```diff
@@ -38,25 +37,8 @@ class WC_Admin_Marketplace_Promotions {
 	 * @return void
 	 */
 	public static function init() {
-		/**
-		 * Filter to suppress the requests for and showing of marketplace promotions.
-		 *
-		 * @since 8.8
-		 */
-		if ( apply_filters( 'woocommerce_marketplace_suppress_promotions', false ) ) {
-			return;
-		}
-
-		register_deactivation_hook( WC_PLUGIN_FILE, array( __CLASS__, 'clear_scheduled_event' ) );
-
-		// Add the callback for our scheduled action.
-		if ( ! has_action( self::SCHEDULED_ACTION_HOOK, array( __CLASS__, 'fetch_marketplace_promotions' ) ) ) {
-			add_action( self::SCHEDULED_ACTION_HOOK, array( __CLASS__, 'fetch_marketplace_promotions' ) );
-		}
-
-		if ( is_admin() ) {
-			add_action( 'init', array( __CLASS__, 'schedule_promotion_fetch' ), 12 );
-		}
+		// A legacy hook that can be triggered by action scheduler.
+		add_action( 'woocommerce_marketplace_fetch_promotions', '__return_true' );
```

Comment:
Clever, I'm glad you thought of that!

Reply:
Thanks! I've changed it a bit in https://github.com/woocommerce/woocommerce/pull/47262/commits/ddd1cb793a69c6cdeeedc5806b2a27c89d585a2f and I wonder if you wouldn't mind reviewing this part?

I'll add the following steps to testing instructions:
1. Go to http://wccore.test:8888/wp-admin/admin.php?page=wc-status&tab=action-scheduler&action=-1&action2=-1&status=pending
2. Check if you have `woocommerce_marketplace_fetch_promotions` in pending actions
3. If yes, run the action
4. Make sure that afterward action `woocommerce_marketplace_fetch_promotions_clear` runs too and that when it's done you see neither `woocommerce_marketplace_fetch_promotions` nor `woocommerce_marketplace_fetch_promotions_clear` in pending actions